Frequent failures of TestUserProvidedCert

**Describe the bug**
While troubleshooting some separate issue with the Kind-based CI jobs, I have observed that the `TestUserProvidedCert` test tend to fail fairly frequently with the following logs:
```
2020-06-10T20:55:17.8005942Z === RUN   TestUserProvidedCert
2020-06-10T20:57:26.4041369Z --- FAIL: TestUserProvidedCert (128.62s)
2020-06-10T20:57:26.4042044Z ##[error]    fixtures.go:55: Creating K8s clientset
2020-06-10T20:57:26.4042787Z ##[error]    fixtures.go:61: Creating 'antrea-test' K8s Namespace
2020-06-10T20:57:26.4043166Z ##[error]    fixtures.go:38: Applying Antrea YAML
2020-06-10T20:57:26.4043553Z ##[error]    fixtures.go:42: Waiting for all Antrea DaemonSet Pods
2020-06-10T20:57:26.4043917Z ##[error]    fixtures.go:46: Checking CoreDNS deployment
2020-06-10T20:57:26.4044615Z ##[error]    security_test.go:192: The CABundle in ConfigMap antrea-ca is valid
2020-06-10T20:57:26.4045701Z ##[error]    security_test.go:209: The CABundle in APIService v1beta1.networking.antrea.tanzu.vmware.com is valid
2020-06-10T20:57:26.4046217Z ##[error]    security_test.go:209: The CABundle in APIService v1beta1.system.antrea.tanzu.vmware.com is valid
2020-06-10T20:57:26.4046630Z ##[error]    security_test.go:222: Expected 3 connected agents, got 0
2020-06-10T20:57:26.4047023Z ##[error]    security_test.go:222: Expected 3 connected agents, got 0
2020-06-10T20:57:26.4047660Z ##[error]    security_test.go:225: Got connections from all 3 antrea-agents
2020-06-10T20:57:26.4048274Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4048896Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4049957Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4050640Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4051269Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4051881Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4052508Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4053114Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4054255Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4054873Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4055493Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4056110Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4056711Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4057326Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4057924Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4058539Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4059137Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4059749Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4060544Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4061147Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4061777Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4063027Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4063764Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4064390Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4064996Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4065623Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4066322Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4066896Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4067450Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4068023Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4068590Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4069148Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4069716Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4070270Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4070962Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4071540Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4072581Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4073162Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4073871Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4074471Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4075220Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4075923Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4076526Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4077141Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4077754Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4078354Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4078965Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4079566Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4080176Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4080769Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4081381Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4081987Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4082586Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4084399Z ##[error]    security_test.go:166: CA bundle doesn't match the expected one, retrying
2020-06-10T20:57:26.4147106Z ##[error]    fixtures.go:196: Deleting 'antrea-test' K8s Namespace
2020-06-10T20:57:26.4147862Z panic: runtime error: invalid memory address or nil pointer dereference [recovered]
2020-06-10T20:57:26.4148070Z 	panic: runtime error: invalid memory address or nil pointer dereference
2020-06-10T20:57:26.4148293Z [signal SIGSEGV: segmentation violation code=0x1 addr=0x358 pc=0x15d9671]
2020-06-10T20:57:26.4148395Z 
2020-06-10T20:57:26.4148533Z goroutine 2384 [running]:
2020-06-10T20:57:26.4148708Z testing.tRunner.func1(0xc0000d2e00)
2020-06-10T20:57:26.4148934Z 	/opt/hostedtoolcache/go/1.13.12/x64/src/testing/testing.go:874 +0x3a3
2020-06-10T20:57:26.4149111Z panic(0x1763460, 0x289d730)
2020-06-10T20:57:26.4149326Z 	/opt/hostedtoolcache/go/1.13.12/x64/src/runtime/panic.go:679 +0x1b2
2020-06-10T20:57:26.4150028Z github.com/vmware-tanzu/antrea/test/e2e.testCert.func1(0xc000a19608, 0xc000a195b8, 0x2)
2020-06-10T20:57:26.4150276Z 	/home/runner/work/antrea/antrea/test/e2e/security_test.go:178 +0x2a1
2020-06-10T20:57:26.4150535Z k8s.io/apimachinery/pkg/util/wait.WaitFor(0xc00086f560, 0xc000a19840, 0xc0008381e0, 0x0, 0x0)
2020-06-10T20:57:26.4150900Z 	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.17.6/pkg/util/wait/wait.go:434 +0x155
2020-06-10T20:57:26.4151144Z k8s.io/apimachinery/pkg/util/wait.pollInternal(0xc00086f560, 0xc000a19840, 0x0, 0x0)
2020-06-10T20:57:26.4151401Z 	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.17.6/pkg/util/wait/wait.go:320 +0x9f
2020-06-10T20:57:26.4151689Z k8s.io/apimachinery/pkg/util/wait.Poll(0x77359400, 0x1e449a9400, 0xc000a19840, 0xc00004c9d0, 0xc00081616e)
2020-06-10T20:57:26.4151944Z 	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.17.6/pkg/util/wait/wait.go:314 +0x4d
2020-06-10T20:57:26.4152491Z github.com/vmware-tanzu/antrea/test/e2e.testCert(0xc0000d2e00, 0xc00024efc0, 0xc00092ed00, 0x889, 0xc00092ed00)
2020-06-10T20:57:26.4152734Z 	/home/runner/work/antrea/antrea/test/e2e/security_test.go:153 +0x1b5
2020-06-10T20:57:26.4153163Z github.com/vmware-tanzu/antrea/test/e2e.TestUserProvidedCert(0xc0000d2e00)
2020-06-10T20:57:26.4153396Z 	/home/runner/work/antrea/antrea/test/e2e/security_test.go:116 +0x5b7
2020-06-10T20:57:26.4153587Z testing.tRunner(0xc0000d2e00, 0x1a3ce98)
2020-06-10T20:57:26.4153966Z 	/opt/hostedtoolcache/go/1.13.12/x64/src/testing/testing.go:909 +0xc9
2020-06-10T20:57:26.4154130Z created by testing.(*T).Run
2020-06-10T20:57:26.4154347Z 	/opt/hostedtoolcache/go/1.13.12/x64/src/testing/testing.go:960 +0x350
2020-06-10T20:57:26.4154738Z FAIL	github.com/vmware-tanzu/antrea/test/e2e	721.891s
2020-06-10T20:57:26.4154880Z FAIL
```

**Versions:**
Antrea v0.7.1

--- a/test/e2e/framework.go
+++ b/test/e2e/framework.go
@@ -739,13 +739,19 @@ func (data *TestData) restartAntreaControllerPod(timeout time.Duration) (*v1.Pod
 		if err != nil {
 			return false, fmt.Errorf("failed to list antrea-controller Pods: %v", err)
 		}
-		for _, pod := range pods.Items {
-			if pod.Status.Phase == v1.PodRunning && pod.DeletionTimestamp == nil {
-				newPod = &pod
-				return true, nil
-			}
+		// Even though the strategy is "Recreate", the old Pod might still be in terminating state when the new Pod is
+		// running as this is deleting a Pod manually, not upgrade.
+		// See https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#recreate-deployment.
+		// So we should ensure there's only 1 Pod and it's running.
+		if len(pods.Items) != 1 {
+			return false, nil
 		}
-		return false, nil
+		pod := pods.Items[0]
+		if pod.Status.Phase != v1.PodRunning || pod.DeletionTimestamp != nil {
+			return false, nil
+		}
+		newPod = &pod
+		return true, nil
 	}); err != nil {
 		return nil, err
 	}

--- a/test/e2e/security_test.go
+++ b/test/e2e/security_test.go
@@ -146,6 +146,9 @@ func testCert(t *testing.T, data *TestData, expectedCABundle string, restartPod 
 		}
 	} else {
 		antreaController, err = data.getAntreaController()
+		if err != nil {
+			t.Fatalf("Error when getting antrea-controller Pod: %v", err)
+		}
 		timeout += 2 * time.Minute
 	}
 

